### PR TITLE
Auth: Add skip_org_role_sync for Okta

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What is this feature?**

This adds the setting `skip_org_role_sync` to Okta oauth provider.

- adds doc
- adds the `skip_org_role_sync` setting to auth.okta section

**Which issue(s) does this PR fix?**:
https://github.com/grafana/grafana-enterprise/issues/4235

Fixes #

**Special notes for your reviewer**:
This does not add tests so far. I have a follow up to this PR to add tests to the `UserInfo` right now, we need to get this into main before code freeze on monday.

### CODE REVIEW

File: pkg/services/login/authinfo.go [hunk 1]
```diff
@@ -25,6 +25,8 @@ const (
 )
 
 func GetAuthProviderLabel(authModule string) string {
```
Comment (eleijonmarck) [inline pkg/services/login/authinfo.go:27]: @IevaVasiljeva  , @gamab or @kalleep perhaps you could answer as to why we do not have Okta in this list?

I was expecting it to be here 🤔

File: pkg/services/login/authinfo.go [hunk 2]
```diff
@@ -25,6 +25,8 @@ const (
...
+	// TODO: future eric, ilove you
```
Comment (eleijonmarck) [inline pkg/services/login/authinfo.go:28]:
```suggestion
```

File: pkg/services/login/authinfo.go [hunk 3]
```diff
@@ -34,6 +34,8 @@ func GetAuthProviderLabel(authModule string) string {
 		return "AzureAD"
 	case "oauth_gitlab":
 		return "GitLab"
+	case "oauth_okta":
```
Comment (eleijonmarck) [inline pkg/services/login/authinfo.go:37]: this is different from previous, as it seems we didn't have okta part of the UI and the backend.

It was "OAuth" previously

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 4]
```diff
@@ -938,6 +938,21 @@ The following table shows the OAuth provider's setting with the default value an
...
+
+When a user logs in the first time, Grafana sets the organization role based on the value specified in `AutoAssignOrgRole`. If you want to manage organization roles, set the `skip_org_role_sync` option to `true`. Okta syncs organization roles and sets Grafana Admins.
+This also impacts `allow_assign_grafana_admin` setting, by not syncing the grafana admin role from GitLab.
+
+> **Note:** There is a separate setting called `oauth_skip_org_role_update_sync` which has a different scope. While `skip_org_role_sync` only applies to the specific OAuth provider, `oauth_skip_org_role_update_sync` is a generic setting that affects all configured OAuth providers.
+
+The following table shows the OAuth provider's setting with the default value and the skip org role sync setting.
+| OAuth Provider | `oauth_skip_org_role_sync_update` | `skip_org_role_sync` | Behavior |
+| --- | --- | --- | --- |
+| Okta | false | false | User organization roles are set with `defaultRole` and cannot be changed |
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/_index.md:951]:
```suggestion
| Okta | false | false | User organization roles are set with `defaultRole` and cannot be changed. |
```

**Review by chri2547 — APPROVED** [review]
One very small nit. Thank you!

**Review by Jguer — COMMENTED** [review]
Missing changes to `defaults.ini` and `sample.ini`?

**Comment by eleijonmarck** [comment]
> Missing changes to `defaults.ini` and `sample.ini`?

well spotted!

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 5]
```diff
@@ -938,6 +938,21 @@ The following table shows the OAuth provider's setting with the default value an
 | GitLab | false | true | User organization roles are set with `defaultRole`, and the organization role can be changed for GitLab synced users. |
 | GitLab | true | true | User organization roles are set with `defaultRole` for GitLab. For other providers, the synchronization is skipped, and the org role can be changed, along with other OAuth provider users' org roles. |
 
+### [auth.okta] skip_org_role_sync
+
+When a user logs in the first time, Grafana sets the organization role based on the value specified in `AutoAssignOrgRole`. If you want to manage organization roles, set the `skip_org_role_sync` option to `true`. Okta syncs organization roles and sets Grafana Admins.
```
Comment (Jguer) [inline docs/sources/setup-grafana/configure-grafana/_index.md:943]:
```suggestion
When a user logs in the first time, Grafana sets the organization role based on the value specified in `AutoAssignOrgRole`. If you want to manage organization roles through Grafana's UI, set the `skip_org_role_sync` option to `true`.
```

Some explana

**Review by Jguer — APPROVED** [review]

**Comment by eleijonmarck** [comment]
apologize! included the commits in the merge

**Review by IevaVasiljeva — APPROVED** [review]
Looks good! I haven't tested it, but the code changes look 👌

Would be good to add a test case for this option.

**Comment by eleijonmarck** [comment]
> Looks good! I haven't tested it, but the code changes look 👌
> 
> Would be good to add a test case for this option.

will for sure! I will look into it after this. Just wanted this to be in for 9.4 if possible 👯

**Pull request merged**